fenzo-core/src/main/java/com/netflix/fenzo/plugins/WeightedAverageFitnessCalculator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.fenzo.plugins;
+
+import java.util.List;
+
+import com.netflix.fenzo.TaskRequest;
+import com.netflix.fenzo.TaskTrackerState;
+import com.netflix.fenzo.VMTaskFitnessCalculator;
+import com.netflix.fenzo.VirtualMachineCurrentState;
+
+/**
+ * A fitness calculator that calculates the weighted average of multiple fitness calculators.
+ */
+public class WeightedAverageFitnessCalculator implements VMTaskFitnessCalculator {
+
+    private final List<WeightedFitnessCalculator> calculators;
+
+    public WeightedAverageFitnessCalculator(List<WeightedFitnessCalculator> calculators) {
+        this.calculators = calculators;
+        if (calculators == null || calculators.isEmpty()) {
+            throw new IllegalArgumentException("There must be at least 1 calculator");
+        }
+        double sum = calculators.stream()
+                .map(WeightedFitnessCalculator::getWeight)
+                .mapToDouble(Double::doubleValue).sum();
+        if (sum != 1.0) {
+            throw new IllegalArgumentException("The sum of the weights must equal 1.0");
+        }
+    }
+
+    @Override
+    public String getName() {
+        return toString();
+    }
+
+    @Override
+    public double calculateFitness(TaskRequest taskRequest, VirtualMachineCurrentState targetVM, TaskTrackerState taskTrackerState) {
+        double totalWeightedScores = 0.0;
+        double totalWeights = 0.0;
+        for (WeightedFitnessCalculator calculator : calculators) {
+            double score = calculator.getFitnessCalculator().calculateFitness(taskRequest, targetVM, taskTrackerState);
```

Comment:
When a fitness function returns 0.0 as its score, what it means is that it doesn't fit at all. This should return 0.0 as the final answer. The current code seems to proceed with the average of the other fitness functions. That would be an incorrect result.

Reply:
This makes sense, will change